scripts/codegen/config.py

```diff
@@ -2,58 +2,57 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-import os
-import sys
 import shlex
-from pathlib import Path
+import re
 
 class ConfigMixin(object):
-    __slots__ = []
-
-    _autoconf = None
-    _autoconf_filename = None
-
-    def _autoconf_assure(self):
-        if self._autoconf is None:
-            autoconf_file = self.cmake_variable("PROJECT_BINARY_DIR", None)
-            if autoconf_file is None:
-                if default == "<unset>":
-                    raise self._get_error_exception(
-                        "CMake variable PROJECT_BINARY_DIR not defined to codegen.", 2)
-                return default
-            autoconf_file = Path(autoconf_file).joinpath('include/generated/autoconf.h')
-            if not autoconf_file.is_file():
-                if default == "<unset>":
-                    default = \
-                    "Generated configuration {} not found/ no access.".format(autoconf_file)
-                return default
-            autoconf = {}
-            with open(str(autoconf_file)) as autoconf_fd:
-                for line in autoconf_fd:
-                    if not line.startswith('#'):
-                        continue
-                    if " " not in line:
-                        continue
-                    key, value = shlex.split(line)[1:]
-                    autoconf[key] = value
-            self._autoconf = autoconf
-            self._autoconf_filename = str(autoconf_file)
-
-    def config_property(self, property_name, default="<unset>"):
-        self._autoconf_assure()
-        property_value = self._autoconf.get(property_name, default)
-        if property_value == "<unset>":
-            raise self._get_error_exception(
-                "Config property '{}' not defined.".format(property_name), 1)
-        return property_value
+    def __init__(self, path_to_kconfig_fragment):
```

Comment:
Mixin classes were new to me. I think I understand them now.

But it is not clear to me how I can achieve what I want to do:

What I really want, is the ability to use ConfigMixin without using the rest of the codegen infrastructure. To my pleasant surprise ConfigMixin was very clean and was easy to decouple
from the rest of codegen.

But I need some way of explicitly passing this path to ConfigMixin. From an object-oriented perspective a constructor seemed natural. But any mechanism will do, I have not able to use
a constructor.

I will now begrudgingly revert back to the original design where every member checks if the 'custom' 'constructor' has been run.

Reply:
> What I really want, is the ability to use ConfigMixin without using the rest of the codegen infrastructure. 

Your version of init is perfect to make it an independent class to be used by you independently of codegen.

Reply:
For future migration to codegen you should have a class variable like in the autoconf case ( _autoconf and _autoconf_filename ) and inititialise using your init function. This can later on be replaced by an initialisation through CMake variables or codegen options.